cfme/tests/test_rest.py

```diff
@@ -374,13 +374,485 @@ def test_vm_add_lifecycle_event(request, setup_a_provider, rest_api, vm, from_de
+    elif request.param == 'roles':
+        return (request.getfuncargvalue("roles_data"), rest_api.collections.roles)
+
+
+def test_add_delete_access_control(rest_api_access_control):
+    """Tests creating and deleting access control entity: ['users', 'groups', 'roles'].
+
+    Prerequisities:
+        * An appliance with ``/api`` available.
+
+    Steps:
+        * access_control = ['users', 'groups', 'roles']
+        * POST /api/{access_control} (method ``add``)
+                for users: ``name``, ```userid`, ``email``
+                for groups: ``description``, ``miq_user_role_id``
+                for roles: ``name``, ``settings``
+        * DELETE /api/{access_control}/<id> -> delete only one entity
+        * DELETE /api/{access_control} <- Insert a JSON with list of dicts containing ``href``s to
+            the entity
+        * Repeat the DELETE query -> now it should return an ``ActiveRecord::RecordNotFound``.
+
+    Metadata:
+        test_flag: rest
+    """
+    entities, api = rest_api_access_control
+    assert "add" in api.action and "delete" in api.action
+
+    for entity in entities:
+        api.action.add(entity)
+
+        wait_for(
+            lambda: api.find_by(name=entity.get("name")),
+            num_sec=180,
+            delay=10,
+        )
+
+    delete_entity = api.find_by(entities[0].get('name'))
+    delete_entity.action.delete()
+    wait_for(
+        lambda: not api.find_by(name=entities[0].get('name')),
+        num_sec=180,
+        delay=10,
+    )
+
+    entities = [entity.id for entity in api]
+    api.delete(entities)
+    with error.expected("ActiveRecord::RecordNotFound"):
+        api.action.delete(entities)
+
+
+def test_edit_access_control(rest_api_access_control):
+    """Tests editing a access_control entity: ['user', 'group', 'role'].
+
+    Prerequisities:
+        * An appliance with ``/api`` available.
+
+    Steps:
+        * access_control = ['users', 'groups', 'roles']
+        * Retrieve list of entities using GET /api/{access_control} , pick the first one
+        * POST /api/{access_control}/<id> (method ``edit``) with the ``name``
+
+    Metadata:
+        test_flag: rest
+    """
+    entities, api = rest_api_access_control
+    assert "edit" in api.action
+
+    try:
+        entity = api[0]
+    except:
+        api.action.add(entities[0])
+        entity = api.find_by(name=(entities[0].get('name') or entities[0].get('description')))
+
+    new_name = "name_{}".format(fauxfactory.gen_alphanumeric())
+
+    entity.action.edit(name=new_name)
+    wait_for(
+        lambda: api.find_by(name=new_name),
+        num_sec=180,
+        delay=10,
+    )
+
+
+@pytest.fixture(scope="module")
+def policies_data():
+    name = fauxfactory.gen_alphanumeric()
+    result = [{
+        "description": "description_{}_{}".format(index, name),
+        "mode": "compliance",
+        "towhat": "Vm",
+    } for index in range(1, 5)]
+
+    return result
```

Comment:
Though this is perfectly valid, I wonder if naming these variables to be a little more telling, rather than```result``` may help someone who is scan reading to code.